app/packs/javascripts/dynamic_elements/topic.js

```diff
@@ -171,6 +171,44 @@ export default class Topic extends ShikiEditable {
         this.$commentsHider.show();
       }
     });
+
+    this._bindReturnClick();
+  }
+
+  _showReturnButton() {
+    if ($('.b-comments').height() > $(window).height()) {
+      $('#return-to-reply').show();
+    }
+  }
+
+  _saveRepliablePosition(commentId) {
+    const repliablePositionKey = 'repliableId_' + window.location.pathname;
+    window.sessionStorage.setItem(repliablePositionKey, commentId);
+  }
+
+  _returnToReplyComment() {
+    const repliablePositionKey = 'repliableId_' + window.location.pathname;
+    const savedId = window.sessionStorage.getItem(repliablePositionKey);
+    if (savedId.length) {
+      $(window).scrollTop(100);
+      $('html, body').animate({
+        scrollTop: $(`#${savedId}`).offset().top
+      }, 0);
```

Comment:
анимация возврата работает как-то неправильно. можно проще
```suggestion
      $.scrollTo($(`#${savedId}`);
```